app/controllers/items_controller.rb

```diff
@@ -0,0 +1,76 @@
+require 'net/http'
+
+class ItemsController < ApplicationController
+
+	def new
+		@item = Item.new
+	end
+
+	def create
+		@item = Item.new
+
+		url = item_params[:item_url]
+
+		if !validate_url(url)
+			flash[:error] = "Please enter a valid Ikea product URL or Article Number!"
+			render :new and return
+		end
+
+		create_item(@item, url, item_params)
+
+		if @item.save
+			flash[:notice] = "Here's your item!"
+			redirect_to @item
+		elsif @item.errors[:item_url][0] == "has already been taken"
+			original_item = Item.find_by(item_url: @item.item_url)
+			redirect_to original_item
+		else
+			# This is here to catch unknown errors so they can be handled better
+			raise StandardError, "Unhandled invalid data in the item creation request!"
+		end
+	end
+
+	def show
+		@item = Item.find(params[:id])
+	end
+
+	def item_params
+		params.require(:item).permit(:item_url)
+	end
+
+	private
+
+	def create_item(item, url, item_params)
```

Comment:
I really like how you've abstracted this work into private methods (beyond the standard `params`). In the next iteration/refactor, think about what parts might make sense to put down on the model. But just wanted to call this out as a really nice step in the right direction with some complex logic!